### ISSUE TITLE
Look up by second level domain.

### ISSUE THREAD

**Issue by developer91234** [keepassxreboot/keepassxc#6519]
This is what all other password fillers do since sites change their login entry-points often. If I have a mail.google.com account, other fillers will fill it on all google sites. Keepass browser will only fill it on mail.google.com (which is no longer used by google or logins). I don't see the use case for looking up by complete domain name.

**Comment by varjolintu** [comment]
Yes, an opposite option for a strict matching would be ideal. The default matching should not be changed.

**Comment by WhyNotHugo** [comment]
This list is very relevant here: https://publicsuffix.org/list/

It's a list of domains where subdomains are considered separate sites (so credentials across them should not be shared).

For example, `github.io` is listed there, so any subdomains of it should be considered different sites, and passwords for one domain should not be suggested for another.

**Comment by dierochade** [comment]
May I ask what is the current state of this. 

I would like to migrate from keepass on desktop (stay with strongbox on iOS :-) and run version 2.7.7.

When I got a ton of database entries like this: URL: https://secure.booking.com
The actual login landing page is:   https://account.booking.com/sign-in?
→The browser extension is unable to match

I thought this was already addressed in [#7988](https://github.com/keepassxreboot/keepassxc/pull/7988), but no matter if I choose activated or deactivated in group settings/browser integration/"omit www subdomain from matching", I get no match in the above described example. 
In Application Settings/Browser integration/ i have not checked "Return only best matching credentials" 

Am I doing it wrong or is this still a problem when migrating with an existing database?

**Comment by droidmonkey** [comment]
Simply add `https://account.booking.com` as an additional url in the entry browser settings.

**Comment by dierochade** [comment]
This is not that easy cause I have several hundred entries that i would need to edit manually.

**Comment by droidmonkey** [comment]
You have several hundred entries that have differing subdomains? That seems unlikely. However, your only option is to hop to it and make the updates as of right now.

**Comment by dierochade** [comment]
I have 1200+ entries, so its just a rough guess. These entries were created with keypass or 1pass. Quite often, the register form and the login form are on different subdomains. In these programms, there was no validation/differentation for subdomains...

So it is not a bad configuration, but just not supported at the moment? Might this have a chance in the future?

I think with passkey support in keepassxc there are  quite some people thinking about migration, but as it will not work (apparantly) they could refrain.

In my personal opinion it would be good if there was an option that the programm matches the whole domain at least if there is no match in the subdomain?

Thank you for the instant support btw.

**Comment by droidmonkey** [comment]
Yes, see my comment above: https://github.com/keepassxreboot/keepassxc/issues/6519#issuecomment-840216037

Passkeys require a new registration so this will not be an issue

**Comment by dierochade** [comment]
Yeah, as you wrote  
> "I could see an application setting that is basically the opposite of return best matching credentials. I'll reopen this with that thought.

i wonder if you still think its a possibility and maybe could implement it. There are users that would definitely appreciate this (me for instance : )

It is true that passkeys need a new registration.  Thats not what I meant. I have to decide to stick with keepass for now or manually update all entries. It is not an option to do them one by one for me, cause it would be a totally annoying experience for months and moreover my partner would not like to have a new software that does not work just because it has some edge feature (very cool!) and looks nicer (it does!)-  instead of the used and working way.

**Comment by droidmonkey** [comment]
I'm in a feature development mood so I'll look into how easy this would be

**Comment by varjolintu** [comment]
This option should be quite easy to make now. The latest versions already implemented a method to get the top level domain.

**Comment by zhangqi222** [comment]
@droidmonkey @phoerious 

I hope this feature can be implemented as soon as possible. When using LastPass, Bitwarden, Edge, or Chrome’s built-in password manager, all of them support matching based on the main domain and ignore subdomains. Even the KeePass client (which I'm currently using) has a toggle for this behavior. I've long wanted to migrate to KeePassXC, but after trying, I found that passwords couldn’t be matched correctly, so I gave up.

I checked the issue log and noticed that this feature has been removed from the “future feature” label. That was really disappointing.

I believe software, as a tool, should make things easier for users. Reducing unnecessary manual operations is one direction toward usability. Developers should work toward making software adapt to users—not force users to adapt to software.

Although version 2.7.10 has improved main domain matching, entries still cannot be matched across different subdomains. For example, if I have a saved entry with the URL translate.google.com, it can match google.com, but it won’t match gmail.google.com. I also saw that wildcard support has been added, but that still requires manually editing each entry, which is a significant workload.

I’ve even considered exporting all entries to CSV and writing a script to batch-remove subdomains in order to achieve the behavior I want. But currently, KeePassXC still preserves the subdomain when saving new entries. If a site changes its structure again in three to five years, I would need to repeat the entire process, increasing the burden on the user.

I imported my data from LastPass into KeePass, and now have over 600 entries, with password records dating back to 2012. Over the years, many of these websites have changed their code and subdomains used for login, but the main domains have remained the same. With just one toggle, I could match passwords without needing to modify any of those 600+ entries. But KeePassXC currently requires users to update every single entry's URL, which is far too much work—and frankly, unacceptable.

If the KeePassXC developers insist on enforcing strict subdomain matching, I sincerely hope you can at least add a toggle like KeePass does—one that allows users to switch between strict (including subdomains) and relaxed (main domain only) matching. That way, even if an entry's URL includes a subdomain, once the toggle is enabled, it could still match any login form under the same main domain.

**Comment by varjolintu** [comment]
> For example, if I have a saved entry with the URL translate.google.com, it can match google.com, but it won’t match gmail.google.com. I also saw that wildcard support has been added, but that still requires manually editing each entry, which is a significant workload.

Just using google.com or account.google.com should solve the issue. Subdomain matching is still needed, in my opinion. The new option for more relaxed matching might be the only solution to your problem.

Also, I'm a bit curious if you have a ten different Google credentials stored with different URLs instead of one per account? Not sure if that's the case, but if so, there's definitely some room for improving the credential management.

> But currently, KeePassXC still preserves the subdomain when saving new entries.

This could also be a new option for the extension: "Save only the top level domain".

**Comment by zhangqi222** [comment]
>  This could also be a new option for the extension: "Save only the top level domain".


This feature is great — it's **simple and effective**.  
**I hope it will be released soon.**  
Thank you!

**Comment by JohnLGalt** [comment]
> Just using google.com or account.google.com should solve the issue.

That is what I ended up doing with my database after bringing it over from KeePass 2.x to KPXC, after using KPXC for a couple of months. It's made for far fewer headaches in the long run, to be fair. 

It took some time to get them all done, yes, but in the end, it was worth using base domains in all my entries as needed (except for a couple that absolutely had to have the FQDN in place).
